packages/hackmcx-api/src/controllers/users.controller.js

```diff
@@ -22,38 +23,49 @@ export async function getUsersByUserId(req, res){
         })
 }
 
-export async function postUser(req, res){
+function userValidation(req){
     const emptyUsername = !req.body.username || /^\s*$/.test(req.body.username);
     const emptyFirstName = !req.body.firstname || /^\s*$/.test(req.body.firstname);
-    const emptyLastName = !req.body.lastname || /^\s*$/.test(req.body.lastname);    
-    const noGivenImageUrl = !req.body.imageUrl;
+    const emptyLastName = !req.body.lastname || /^\s*$/.test(req.body.lastname);
+    const emptyPassword = !req.body.password || /^\s*$/.test(req.body.password);  
     const notPicture = !req.body.imageUrl || !/(https?:\/\/.*\.(?:png|jpg|gif|svg))/i.test(req.body.imageUrl);
+    const noGivenImageUrl = !req.body.imageUrl;
 
     if(emptyUsername){
```

Comment:
I think these validation checks can be factored out, and maybe applied elsewhere. 

Eventually we might want to look at just adding some reasonable exception handling and breaking all this validation out of the controller.